src/mprisplayer.cpp

```diff
@@ -82,24 +83,19 @@ MprisPlayerPrivate::~MprisPlayerPrivate()
     }
 }
 
-MprisPlayer *MprisPlayerPrivate::parent() const
-{
-    return static_cast<MprisPlayer *>(QObject::parent());
-}
-
 void MprisPlayerPrivate::quit()
 {
-    Q_EMIT parent()->quitRequested();
+    Q_EMIT q_ptr->quitRequested();
 }
 
 void MprisPlayerPrivate::raise()
 {
-    Q_EMIT parent()->raiseRequested();
+    Q_EMIT q_ptr->raiseRequested();
 }
 
 qlonglong MprisPlayerPrivate::position() const
 {
-    return static_cast<MprisPlayer *>(parent())->position() * 1000;
+    return static_cast<MprisPlayer *>(q_ptr)->position() * 1000;
```

Comment:
This `static_cast` seems redundant. Could remove it while at it.

Reply:
Duh, already did while going through the changes on the PR but looks like I ended up failing pushing the latest version.